iModelCore/ecobjects/src/ECSchema.cpp

```diff
@@ -3211,6 +3211,37 @@ ECSchemaPtr SearchPathSchemaFileLocater::_LocateSchema(SchemaKeyR key, SchemaMat
     return schemaOut;
     }
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+ECSchemaPtr StringSchemaLocater::_LocateSchema(SchemaKeyR key, SchemaMatchType matchType, ECSchemaReadContextR schemaContext)
+    {
+    // Get cached version of the schema
+    ECSchemaPtr schemaOut = schemaContext.GetFoundSchema(key, matchType);;
+    if (schemaOut.IsValid())
+        {
+        return schemaOut;
+        }

```

Comment:
The schema locater should not look in the context.  It should just load the schema on request.  The schema context will check it's cache before calling the locaters registered to the context

Reply:
This call is wrong, however, the other locaters check the context before they attempt to load a schema. Like, if Schema XY version 1.0.0 matchType "Latest" is requested, and our locater can provide version 3.0.0, it can use GetFoundSchema(<for 3.0.0>) before actually attempting to load that one.
I also first thought this is unnecessary, but the other locaters seem to do this as well.

Reply:
The SearchPathSchemaFileLocater is overly complex. 

If version 3.0.0 was already in the cache then it would have been loaded from the cache before this locater was called.  I think it's possible that this code exists in the SearchPathSchemaFileLocater but is no longer necessary.  This code has gone through a lot of permutations.  I went back a bit in the history and this has been in the code since very early, I didn't dig into it enough to understand how things worked at that point but it was a lot different.  Let's try without it.

Reply:
The updated schema locator no longer makes a call to the schema context method.